app/pages/unsubscribe.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { Component, PropTypes } from 'react';
+import Auth from 'panoptes-client/lib/auth';
+
+
+class UnsubscribeFromEmails extends Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.renderForm = this.renderForm.bind(this);
+    this.renderIconFeedback = this.renderIconFeedback.bind(this);
+    this.renderTextFeedback = this.renderTextFeedback.bind(this);
+    this.renderTokenFlowFeedback = this.renderTokenFlowFeedback.bind(this);
+    this.state = {
+      emailSuccess: false,
+      emailError: false,
+      emailIsValid: false,
+      emailValue: '',
+      inProgress: false,
+    };
+  }
+  componentDidMount() {
+    return this.handleChange;
+  }
+
+  handleChange(event) {
+    const emailIsValid = event.target.checkValidity();
+    const emailValue = event.target.value || event.target.defaultValue;
```

Comment:
If both do exist, and they have different values, which one should you unsubscribe? I'd go with the input value.